pkg/controllers/user/alert/watcher/node.go

```diff
@@ -69,13 +69,8 @@ func (w *NodeWatcher) watchRule() error {
 		if alert.Spec.TargetNode.NodeName != "" {
 			parts := strings.Split(alert.Spec.TargetNode.NodeName, ":")
 			id := parts[1]
-			newNode, err := w.nodeLister.Get("", id)
```

Comment:
id := parts[1]
may be panic, if there are invalid format and the array length is 0.

Reply:
I added the following logic to check:
```
if len(parts) != 2 {
  continue
}
```